### PR TITLE
feat(site): add an organization switcher to the user menu

### PR DESCRIPTION
**Pull request by aslilac**
Screenshot 2024-05-13 at 4 05 59 PM

### CODE REVIEW

File: site/src/contexts/auth/RequireAuth.tsx [hunk 1]
```diff
@@ -67,7 +67,7 @@ export const RequireAuth: FC = () => {
 };
 
 type RequireKeys<T, R extends keyof T> = Omit<T, R> & {
-  [K in keyof Pick<T, R>]: NonNullable<T[K]>;
+  [K in keyof Pick<T, R>]-?: NonNullable<T[K]>;
```
Comment (BrunoQuaresma) [inline site/src/contexts/auth/RequireAuth.tsx:70]: What `-?` does?

**Review by BrunoQuaresma — COMMENTED** [review]
I downloaded the branch locally but I can't see the teams getting listed in the dropdown - or at least the default one.

Screenshot 2024-05-15 at 09 54 56

How can I test this feature?

**Comment by aslilac** [comment]
You need to run with `CODER_EXPERIMENTS=multi-organization` _and_ click the secret button that that enables in your user settings, and refresh your page because the `localStorage` change doesn't get picked up by React, but it's not worth fixing because the button is super temporary.

**Comment by aslilac** [comment]
_and_ you'll need to manually hit the backend if you want to click a different org, _and_ it'll probably break everything if you do. Trying to hack this out one piece at a time. 😅

File: site/src/contexts/auth/RequireAuth.tsx (hunk 1 above)
Comment (aslilac) [reply site/src/contexts/auth/RequireAuth.tsx:70]: "make the mapped key non-optional". it's weird but `NonNullable` is not enough on it's own. 🙃

**Review by BrunoQuaresma — APPROVED** [review]
I'm going to be off tomorrow so I'm going to approve this since the code looks good and it is under experimental 👍

**Pull request merged**